### PR TITLE
User facing wallet wipe

### PR DESCRIPTION
**Pull request by BrodyHughes**
Changes:
- Moving Wallets & Keys to a top level settings item
- Surfacing Wallet wipe to be user facing
- Whole flow is keyboard navigable
- The basic flow and functionality are there. Will create a follow up PR adding this to the individual wallet details section's like Daniel has requested.

Flow:

<video src="https://github.com/rainbow-me/browser-extension/assets/41711440/db4e9b1d-aeaa-404e-af21-cb8b23bcec58" controls="controls" style="max-width: 730px;"></video>

### CODE REVIEW

File: src/entries/popup/components/WarningInfo/WalletWipeWarningInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+              color={buttonEnabled ? 'red' : 'fill'}
+              height="44px"
+              variant="flat"
+              width="full"
+              symbol={buttonEnabled ? 'trash' : 'arrow.up.circle.fill'}
+              blur="26px"
+              onClick={onProceed}
+              disabled={!buttonEnabled}
+            >
+              {buttonEnabled ? 'Wipe Wallets' : 'Complete Check Above'}
```
Comment (estebanmino) [inline src/entries/popup/components/WarningInfo/WalletWipeWarningInfo.tsx:150]: to lang file?

File: static/json/languages/en_US.json [hunk 2]
```diff
@@ -222,6 +222,13 @@
           "warning_4": "Rainbow Support will never ask you for your Secret Recovery Phrase.",
           "show": "Show Recovery Phrase",
           "copied": "Recovery Phrase Copied"
+        },
+        "wipe_wallets": {
+          "delete": "Delete Wallets & Clear Keychain",
+          "acknowlegement": "I have read and ackowledge the above info.",
```
Comment (estebanmino) [inline static/json/languages/en_US.json:228]: is ackowledge a word?

**Comment by estebanmino** [comment]
@BrodyHughes is there any ticket for this?

File: src/entries/popup/pages/settings/walletsAndKeys/walletsAndKeys.tsx [hunk 3]
```diff
@@ -341,6 +347,33 @@ export const WalletsAndKeys = () => {
...
+            rightComponent={
+              <Symbol
+                size={16}
+                color="red"
+                weight="medium"
+                symbol="exclamationmark.triangle"
+              />
+            }
+            titleComponent={
+              <MenuItem.Title text={t('wipe_wallets.delete')} color="red" />
```
Comment (estebanmino) [inline src/entries/popup/pages/settings/walletsAndKeys/walletsAndKeys.tsx:372]: should we hide this flow when there are no imported wallets?

Comment (BrodyHughes) [reply src/entries/popup/pages/settings/walletsAndKeys/walletsAndKeys.tsx:372]: @DanielSinclair thoughts? I could see it being useful for people who have several watched wallets and want to reset.

**Comment by BrodyHughes** [comment]
> @BrodyHughes is there any ticket for this?

no, it's just a roadmap item: https://www.notion.so/rainbowdotme/Reset-Wallet-Delete-Seeds-Settings-886821e19fd8490aa8b9059ec4ce5031

File: src/entries/popup/components/WarningInfo/WalletWipeWarningInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+}
+
+export default function WarningInfo({
+  iconAndCopyList,
+  onProceed,
+  testId,
+}: WarningInfoProps) {
+  const [acknowledgeCheck, setacknowledgeCheck] = useState(false);
+
+  const buttonEnabled = useMemo(() => acknowledgeCheck, [acknowledgeCheck]);
```
Comment (greg-schrammel) [inline src/entries/popup/components/WarningInfo/WalletWipeWarningInfo.tsx:41]: this useMemo does nothing
```suggestion
  const buttonEnabled = !!acknowledgeCheck;
```

File: src/entries/popup/components/WarningInfo/WalletWipeWarningInfo.tsx [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+  onProceed: () => void;
+  testId?: string;
+}
+
+export default function WarningInfo({
+  iconAndCopyList,
+  onProceed,
+  testId,
+}: WarningInfoProps) {
+  const [acknowledgeCheck, setacknowledgeCheck] = useState(false);
```
Comment (greg-schrammel) [inline src/entries/popup/components/WarningInfo/WalletWipeWarningInfo.tsx:39]: camel case
```suggestion
  const [acknowledgeCheck, setAcknowledgeCheck] = useState(false);
```

File: src/entries/popup/pages/settings/walletsAndKeys/wipeWalletWarning.tsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+  }, [address, setCurrentAddress]);
+
+  const closePasswordPrompt = useCallback(
+    async () => setShowEnterPassword(false),
+    [],
+  );
+
+  const handleWipeWallet = useCallback(async () => {
+    await wallet.wipe();
+    await updateState();
```
Comment (greg-schrammel) [inline src/entries/popup/pages/settings/walletsAndKeys/wipeWalletWarning.tsx:64]: if `updateState` is not used anywhere else I vote for inlining it here

**Comment by DanielSinclair** [comment]
@BrodyHughes Looking solid 👌 

Some ideas:
- I think `Reset Rainbow` might be a slightly more intuitive title (though maybe `Reset Rainbow & Clear Wallets` is more explanatory), and I think a `>` arrow like this would be a bit more consistent with the design language
Screenshot 2024-01-04 at 1 42 09 AM

- We should add the password prompt before you get to the review screen to match the View Secret Recovery Phrase behavior
Screenshot 2024-01-04 at 2 02 11 AM

- I think we should add another bullet point to the front that explains "Reseting Rainbow will clear your settings, reset your password, and permanently remove all of your wallets". Then the next point point should reiterate what clearing them means.

- Let's use `Secret Recovery Phrase` on the current 2nd bullet point and add a learn more link. We'll need to push up an article, so let's use this placeholder `https://rainbow.me/support/extension/backing-up-your-wallets`.

- For the last bullet point, let's add something that mentions assets or funds in relation to the backed up wallets

- Keeping consistent, I think the destructive confirm button in this particular flow should be `Reset Rainbow`

- I like the checkbox but I think we should also add another prompt at the end (now that the password is front-loaded) with a destructive Reset button
Screenshot 2024-01-04 at 1 56 11 AM

- Behavior question: when the user does this from the pop-over, do we open a New Tab with the Onboarding page? If not, I think we should do that similar to how we trigger that page to open upon new install

- I think we should also expose this flow in the drill-down for each Wallet Group. I think the only changes would be:
    1. Entry button title for seeds should be "Remove Recovery Phrase"
    2. Other "Reset Rainbow" strings would be "Remove Recovery Phrase" or "Remove" (i.e. prompt button) and the list strings would be slightly different
    3. For Hardware Wallets, we'd use the language "Unpair Hardware Wallet" or "Unpair" and the list strings would be slightly different

Screenshot 2024-01-04 at 1 57 58 AM

File: src/entries/popup/components/IconAndCopyList.tsx/IconAndCopyList.tsx [hunk 7]
```diff
@@ -39,6 +46,16 @@ export function IconAndCopyList({
             </Box>
             <Text size="14pt" weight="medium" color="label">
               {copy}
+              {link && (
+                <TextLink
+                  color="blue"
+                  onClick={() =>
+                    goToNewTab({ url: 'https://learn.rainbow.me/' })
```
Comment (estebanmino) [inline src/entries/popup/components/IconAndCopyList.tsx/IconAndCopyList.tsx:53]: shouldn't you be using `href` prop for this link? feels like this is adding a hardcoded link to all `IconAndCopyList` components that may not even be related to that link

**Review by estebanmino — APPROVED** [review]
didn't test it but code lgtm

**Review by brunobar79 — APPROVED** [review]
Looks great!

File: src/entries/popup/components/IconAndCopyList.tsx/IconAndCopyList.tsx [hunk 8]
```diff
@@ -39,6 +42,11 @@ export function IconAndCopyList({
             </Box>
             <Text size="14pt" weight="medium" color="label">
               {copy}
+              {onClickLink && (
+                <TextLink color="blue" onClick={onClickLink}>
```
[the remark this answers is not in the document]
Comment (BrodyHughes) [reply src/entries/popup/components/IconAndCopyList.tsx/IconAndCopyList.tsx:46]: yeah good call.

**Pull request merged**